src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerAnalyzer.cs

```diff
@@ -81,13 +117,19 @@ protected override void AddMatches(ArrayBuilder<Match<TStatementSyntax>> matches
             if (initializer != null)
             {
                 var firstInit = _syntaxFacts.GetExpressionsOfObjectCollectionInitializer(initializer).First();
+
+                // if we have an object creation, and it *already* has an initializer in it (like `new T { { x, y } }`)
+                // this can't legally become a collection expression.
+                if (_analyzeForCollectionExpression && this.IsComplexElementInitializer(firstInit))
+                    return false;
```

Comment:
needed to make this bool returning as we cannot use the "empty result means failure" approach for 'collection expressions'.  For collection expressions an empty result is fine and means "convert to `[]`".